packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -72,31 +74,66 @@ class CodeSnippetDisplay extends React.Component<ICodeSnippetDisplayProps> {
       const documentWidget = widget as DocumentWidget;
       const fileEditor = (documentWidget.content as FileEditor).editor;
 
-      // Wrap in code block if inserting snippet into a markdown file
       if (PathExt.extname(widget.context.path) === '.md') {
+        // Wrap snippet into a code block when inserting it into a markdown file
         fileEditor.replaceSelection(
           '```' + snippet.language + '\n' + snippetStr + '\n```'
         );
+
+        // TODO: Change how python file editor is checked below to be more generic
+      } else if (widget instanceof PythonFileEditor) {
+        this.verifyLanguage(snippet.language, 'python', fileEditor, snippetStr);
       } else {
-        // Allow code snippet insertion independent of file extension
         fileEditor.replaceSelection(snippetStr);
       }
     } else if (widget instanceof NotebookPanel) {
       const notebookWidget = widget as NotebookPanel;
       const notebookCellEditor = (notebookWidget.content as Notebook).activeCell
         .editor;
-      notebookCellEditor.replaceSelection(snippetStr);
+      const kernelLanguage: string =
+        notebookWidget.context.sessionContext.kernelPreference.language;
+      this.verifyLanguage(
+        snippet.language,
+        kernelLanguage,
+        notebookCellEditor,
+        snippetStr
+      );
     } else {
-      this.showErrorDialog('Code snippet insert failed: Unsupported widget');
+      this.showErrDialog('Code snippet insert failed: Unsupported widget');
     }
   }
 
-  private showErrorDialog = (
-    errorMsg: string
+  private verifyLanguage = async (
+    snippetLanguage: string,
+    editorLanguage: string,
+    editor: CodeEditor.IEditor,
+    code: string
+  ): Promise<void> => {
+    if (snippetLanguage !== editorLanguage) {
+      const result = await this.showWarnDialog(editorLanguage);
+      result.button.accept && editor.replaceSelection(code);
```

Comment:
Though this shorthand is valid, I think we should use a full if statement for better readability by those that don't know js shorthands